diff -up libxml2-2.6.32/python/libxml.c.orig libxml2-2.6.32/python/libxml.c
--- libxml2-2.6.32/python/libxml.c.orig	2007-01-09 23:22:52.000000000 +0200
+++ libxml2-2.6.32/python/libxml.c	2009-09-07 15:45:40.000000000 +0300
@@ -2500,11 +2500,11 @@ libxml_last(ATTRIBUTE_UNUSED PyObject * 
         case XML_DTD_NODE:
             res = cur->last;
             break;
-        case XML_ATTRIBUTE_NODE:{
+        case XML_ATTRIBUTE_NODE: {
                 xmlAttrPtr attr = (xmlAttrPtr) cur;
-
                 res = attr->last;
             }
+            break;
         default:
             res = NULL;
             break;
